src/Events/Appointments/AppoinmentSetEvent.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ *
+ * @package      OpenEMR
+ * @link               https://www.open-emr.org
+ *
+ * @author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c) 2021 Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ */
+
+namespace OpenEMR\Events\Appointments;
+
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event object for know what type of appointment has been set on the calendar
+ *
+ * @package OpenEMR\Events
+ * @subpackage Appointments
+ *
+ */
+class AppoinmentSetEvent extends Event
+{
+    /**
+     * This event is triggered after a new patient appointment has been scheduled
+     */
+    const EVENT_HANDLE = 'appointment.set';
+
+    /**
+     * @var
+     */
+    public $post;
+
+    /**
+     * @var
+     */
+    public $eid;
```

Comment:
prob would make this private and then make a function to set it or include it within the constructor (would defer this to @adunsulag though since he's the event listener expert :) )

Reply:
So the event consumer will need access to this variable so it should be accessible but I would prefer it be done via a getter instead of the core data being public.  It allows the underlying variable to change while still keeping the method signature the same.  These events will be some of the hardest pieces to change in the future due to developers reliance on them so we have to make sure we keep their API contracts consistent as possible.  I'm not sure if we've been very consistent with event objects elsewhere but we should probably start doing so. 

I don't have a ton of energy over the getters/public but it is better practice that it is done via getter.  Whatever we go with will not be able to changed easily in the future so we should probably on all future code reviews force events to use getters for access to their data.